### PR TITLE
Update order on gem

### PR DESCRIPTION
**Pull request by Dujota**
Updates to the existing work already done on the first iteration of the orders added to the spire gem

### CODE REVIEW

File: lib/spire/order.rb [hunk 1]
```diff
@@ -1,94 +1,218 @@
...
+      :terms_text, :freight, :taxes, :subtotal, :subtotal_ordered, :discount,
+      :total_discount, :total, :total_ordered, :gross_profit, :items, :created_by,
+      :modified_by, :created, :modified, :background_color,
       readonly: [
-        :id
+        :created_by, :modified_by, :created, :modified, :order_no
       ]
 
-    validates_presence_of :customer, :items
+    validates_acceptance_of :id, :customer, :address, :shipping_address, :items
```
Comment (caseyli) [inline lib/spire/order.rb]: @Dujota I feel like acceptance is not being properly used here. I feel like presence still makes sense.

File: lib/spire/order.rb [hunk 2]
```diff
@@ -1,94 +1,218 @@
...
+      status: 'status',
+      type: 'type',
+      hold: 'hold',
+      order_date: 'orderDate',
       address: 'address',
-      shipping_address: "shippingAddress",
+      shipping_address: 'shippingAddress',
+      customer_po: 'shippingPO',
+      fob: 'fob',
+      terms_code: "termsCode",
```
Comment (thejackwu) [inline lib/spire/order.rb]: might be keep it a single quote?

File: lib/spire/order.rb [hunk 3]
```diff
@@ -1,94 +1,218 @@
...
+      order_no: 'orderNo',
       customer: 'customer',
+      status: 'status',
+      type: 'type',
+      hold: 'hold',
+      order_date: 'orderDate',
       address: 'address',
-      shipping_address: "shippingAddress",
+      shipping_address: 'shippingAddress',
+      customer_po: 'shippingPO',
```
Comment (caseyli) [inline lib/spire/order.rb]: @Dujota I feel like these should be called the same. Whatever the API uses, whether it be

```shipping_po: 'shippingPO'```

or

```customer_po: 'customerPO'```

**Review by caseyli — COMMENTED** [review]
@Dujota a couple of comments to address.

File: lib/spire/order.rb [hunk 4]
```diff
@@ -1,94 +1,218 @@
...
+      # Create a new item and save it on Spire.
       #
       # @param [Hash] options
-      #
-      # @raise [Spire::Error] if the item could not be created.
+      # @option options [Hash] :customer the spire customer who will be associated with the order
+      # @option options [Hash] :address this is the billing address for the customer
+      # @option options [Hash] :shipping_address this is the shipping address that the order will be sent to (defaults to the billing address if none provided)
+      # @option options [Array] :items this is an array of hashes that will accept an inventory item that will have a hash example input from the web client: items: [ { "inventory": {"id": 123} } ]
+      # this array can also accept a desctiption and comment object that will create a comment on the order itself ex: items: [{"description":"MAKE COMMENT THRU API","comment":"MAKE COMMENT THRU API"}]
```
Comment (thejackwu) [inline lib/spire/order.rb:136]:
```suggestion
      # this array can also accept description and comment objects that will create a comment on the order itself ex: items: [{"description":"MAKE COMMENT THRU API","comment":"MAKE COMMENT THRU API"}]
```

File: lib/spire/order.rb [hunk 5]
```diff
@@ -1,94 +1,218 @@
...
+          'items' => options[:items],
         )
       end
     end
 
+    # Update the fields of an order.
+    #
+    # Supply a hash of string keyed data retrieved from the Spire API representing an order.
+    #
+    # Note that this this method does not save anything new to the Spire API,
```
Comment (thejackwu) [inline lib/spire/order.rb]:
```suggestion
    # Note that this method does not save anything new to the Spire API,
```

File: lib/spire/order.rb (hunk 3 above)
Comment (Dujota) [reply lib/spire/order.rb]: small typo, should be customerPO

File: lib/spire/order.rb (hunk 1 above)
Comment (Dujota) [reply lib/spire/order.rb]: weird, not sure how that snuck in there, good call though

File: lib/spire/order.rb (hunk 2 above)
Comment (Dujota) [reply lib/spire/order.rb]: yup good catch

File: lib/spire/order.rb (hunk 4 above)
Comment (Dujota) [reply lib/spire/order.rb:136]: this is singular as it is one description and one comment object rather than a collection

**Pull request merged**